p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/transport/RemoteArtifactRepositoryManagerAgentFactory.java

```diff
@@ -45,14 +49,33 @@ public Object createService(IProvisioningAgent agent) {
 	}
 
 	private boolean getDisableP2MirrorsConfiguration() {
-		String key = "tycho.disableP2Mirrors";
-		String value = System.getProperty(key);
-		if (value != null) {
-			logger.info("Using " + key
+		String deprecatedKey = "tycho.disableP2Mirrors";
+		String deprecatedValue = System.getProperty(deprecatedKey);
```

Comment:
its seems a good idea to refactor this and the follogiwn check into a method that:

1. first check for the property in the System.getProperty
2. then if we have a session
3. if session the use system/user properties from there

Reply:
Is this the correct order? Should user property not have priority over system session property and system property?

Reply:
I updated with your suggestion, let me know if I should change the order or not